vyper/semantics/analysis/local.py

```diff
@@ -173,9 +173,13 @@ def __init__(
         self.func = fn_node._metadata["type"]
         self.annotation_visitor = StatementAnnotationVisitor(fn_node, namespace)
         self.expr_visitor = _LocalExpressionVisitor()
+
+        # allow internal function params to be mutable
+        location = DataLocation.MEMORY if self.func.is_internal else DataLocation.CALLDATA
+        is_immutable = False if self.func.is_internal else True
```

Comment:
i don't really like repeating branches in general. here i think we can get rid of it with either squashing all into a single ternary like so
```suggestion
        # allow internal function params to be mutable
        location, is_immutable = (DataLocation.MEMORY, False) if self.func.is_internal else (DataLocation.CALLDATA, True)
```
or two assignments per branch
```suggestion
        if self.func.is_internal:
            location = DataLocation.MEMORY
            is_immutable = False
        else:
            location = DataLocation.CALLDATA
            is_immutable = True
```